rc.js

```diff
@@ -4,49 +4,32 @@ var detectLibc = require('detect-libc')
 var napi = require('napi-build-utils')
 
 var env = process.env
-
 var libc = env.LIBC || (detectLibc.isNonGlibcLinux && detectLibc.family) || ''
 
-// Get `prebuild-install` arguments that were passed to the `npm` command
-if (env.npm_config_argv) {
-  var npmargs = ['prebuild', 'compile', 'build-from-source', 'debug', 'verbose']
-  try {
-    var npmArgv = JSON.parse(env.npm_config_argv).cooked
-    for (var i = 0; i < npmargs.length; ++i) {
-      if (npmArgv.indexOf('--' + npmargs[i]) !== -1) {
-        process.argv.push('--' + npmargs[i])
-      }
-      if (npmArgv.indexOf('--no-' + npmargs[i]) !== -1) {
-        process.argv.push('--no-' + npmargs[i])
-      }
-    }
-    if ((i = npmArgv.indexOf('--download')) !== -1) {
-      process.argv.push(npmArgv[i], npmArgv[i + 1])
-    }
-  } catch (e) { }
-}
-
 // Get the configuration
 module.exports = function (pkg) {
   var pkgConf = pkg.config || {}
-  var sourceBuild = env.npm_config_build_from_source
-  var buildFromSource = sourceBuild === pkg.name || sourceBuild === 'true'
+
+  // TODO: remove compile and prebuild aliases?
+  var buildFromSource = env.npm_config_build_from_source || env.npm_config_compile
```

Comment:
I agree we could rely on `npm_config_build_from_source` only here.

Reply:
OK, will make a second PR for that once this lands.